cloudtools/connect.py

```diff
@@ -54,13 +55,63 @@ def main(args):
     with open(os.devnull, 'w') as f:
         check_call(cmd, stdout=f, stderr=f)
 
-    # open Chrome with SOCKS proxy configuration
+    if sys.platform.startswith('linux'):
+        names = ['chromium-browser', 'chromium', 'google-chrome']
+        # attempt to find a chrome/chromium browser in the user's PATH
+        for name in names:
+            browser = which(name)
+            if browser is not None:
+                break
+        if browser is None:
+            raise Exception('could not find a chromium browser. searched for {}'.format(names))
+    else:
+        browser = r'/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
+
+    # open Chrome/Chromium with SOCKS proxy configuration
     cmd = [
-        r'/Applications/Google Chrome.app/Contents/MacOS/Google Chrome',
+        browser,
         'http://localhost:{}'.format(connect_port),
         '--proxy-server=socks5://localhost:{}'.format(args.port),
         '--host-resolver-rules=MAP * 0.0.0.0 , EXCLUDE localhost',
         '--user-data-dir=/tmp/'
     ]
     with open(os.devnull, 'w') as f:
         Popen(cmd, stdout=f, stderr=f)
+
+
+def which(cmd):
+    """An almost verbatim copy of python 3.3 and later's shutil.which(), used to provide
```

Comment:
does this make cloudtools subject to cpython's license? @cseed?

Reply:
could also call out to `which` potentially?

Reply:
There is no guarantee that a binary called `which` can be found on the system (I've used such systems). And dan's point of supporting Windows is valid and I would like to do so if possible. If I can't enumerate PATH, then I will find a different way.